build/linux/installer/scripts/tomlparser-agent-config.rb

```diff
@@ -131,6 +137,40 @@ def populateSettingValuesFromConfigMap(parsedConfig)
           puts "Using config map value: NODES_EMIT_STREAM_BATCH_SIZE = #{@nodesEmitStreamBatchSize}"
         end
       end
+      # fbit config settings
+      fbit_config = parsedConfig[:agent_settings][:fbit_config]
+      if !fbit_config.nil?
+        fbitFlushInterval = fbit_config[:FBIT_SERVICE_FLUSH_INTERVAL]
+        if !fbitFlushInterval.nil? && is_number?(fbitFlushInterval) && fbitFlushInterval.to_i > 0
+          @fbitFlushInterval = fbitFlushInterval.to_i
+          puts "Using config map value: FBIT_SERVICE_FLUSH_INTERVAL = #{@fbitFlushInterval}"
+        end
+
+        fbitTailBufferChunkSize = fbit_config[:FBIT_TAIL_BUFFER_CHUNK_SIZE]
+        if !fbitTailBufferChunkSize.nil? && is_number?(fbitTailBufferChunkSize) && fbitTailBufferChunkSize.to_i > 0
+          @fbitTailBufferChunkSize = fbitTailBufferChunkSize.to_i
+          puts "Using config map value: FBIT_TAIL_BUFFER_CHUNK_SIZE = #{@fbitTailBufferChunkSize}"
+        end
+
+        fbitTailBufferMaxSize = fbit_config[:FBIT_TAIL_BUFFER_MAX_SIZE]
+        if !fbitTailBufferMaxSize.nil? && is_number?(fbitTailBufferMaxSize) && fbitTailBufferMaxSize.to_i > 0 
+          if fbitTailBufferMaxSize.to_i >= @fbitTailBufferChunkSize
+            @fbitTailBufferMaxSize = fbitTailBufferMaxSize.to_i
+            puts "Using config map value: FBIT_TAIL_BUFFER_MAX_SIZE = #{@fbitTailBufferMaxSize}"
+          else
+            @fbitTailBufferMaxSize = @fbitTailBufferChunkSize
+            puts "config::warn: FBIT_TAIL_BUFFER_MAX_SIZE should be greater or equal to value of FBIT_TAIL_BUFFER_CHUNK_SIZE. Using FBIT_TAIL_BUFFER_MAX_SIZE = #{@fbitTailBufferMaxSize} since provodided config value not valid"
+          end
+        end
+         # in scenario - FBIT_TAIL_BUFFER_MAX_SIZE provided but not FBIT_TAIL_BUFFER_CHUNK_SIZE and vice versa
```

Comment:
The current behavior is to not set the values in case they are not provided. Why are we setting chunk to max and max to chunk if either is not provided?
Essentially the customer would want to see the exact configuration settings they are providing using configmap.

Reply:
I think the logic should remain the same as here. https://github.com/microsoft/Docker-Provider/blob/ci_dev/build/common/installer/scripts/td-agent-bit-conf-customizer.rb

Reply:
> The current behavior is to not set the values in case they are not provided. Why are we setting chunk to max and max to chunk if either is not provided?
> Essentially the customer would want to see the exact configuration settings they are providing using configmap.

> Buffer_Chunk_Size and Buffer_Chunk_Max_Size are tightly coupled. as pointed out  here
https://fluentbit.io/documentation/0.13/input/tail.html, default value of Buffer_Max_Size is the same as Buffer_Chunk_Size.
https://github.com/fluent/fluent-bit/blob/master/plugins/in_tail/tail_config.c#L165
This is to ensure no undesired effects in case customer provides one value but not other value. If the customer provides both the values and those will be honored as long as meet this criterion buffer_max_size must be >= buffer_chunk.

Reply:
1. If the value is not set for Buffer_Max_Size fluentbit already takes the default value as the Buffer chunk size.
2. If the value is not set for Buffer_Chunk_Size fluentbit takes 32k as the default. Why should we set this to Buffer max size if it is not provided?

Reply:
> 1. If the value is not set for Buffer_Max_Size fluentbit already takes the default value as the Buffer chunk size.
>   where do you see this? can you please point me the code or doc  so that i can validate before removing this additional code ? if it does what you say, then I can safely remove.

> 2. If the value is not set for Buffer_Chunk_Size fluentbit takes 32k as the default. Why should we set this to Buffer max size if it is not provided?
>  To avoid undesired effects when Buffer_Chunk_Size is larger than default max size. If you prove that Buffer_Max_Size will be set same as Buffer_Chunk_Size  and vice versa then I am happy to remove this piece of the code though it does same what you are describing here.

Reply:
1. It is the same documentation for tail plugin you shared earlier - https://fluentbit.io/documentation/0.13/input/tail.html
Look under the table for the settings.
2. In your code "if @fbitTailBufferMaxSize > 0 && @fbitTailBufferChunkSize == 0
          @fbitTailBufferChunkSize = @fbitTailBufferMaxSize"
You are setting the chunk size to max size if it is not set. This is also in the same tail documentation.

Reply:
> 1. It is the same documentation for tail plugin you shared earlier - https://fluentbit.io/documentation/0.13/input/tail.html
>    Look under the table for the settings.
> 2. In your code "if @fbitTailBufferMaxSize > 0 && @fbitTailBufferChunkSize == 0
>    @fbitTailBufferChunkSize = @fbitTailBufferMaxSize"
>    You are setting the chunk size to max size if it is not set. This is also in the same tail documentation.

>  The doc only talking out the default value. do you know if buf_max_size will be buf_chunk_size if not proided? as I pointed out here, I think, if the value of the buf_chunk_size   is greater than buf_max_size  then the fbit will fail with an error https://github.com/fluent/fluent-bit/blob/master/plugins/in_tail/tail_config.c#L165. Let me do quick validation and see what's behavior.

Reply:
In the changes I dont see code where you are checking if chunk size is greater than max size and setting chunk size to max size. The only change I see is if chunk size is not set, it is being set to max, which is not clear why. 
And yes the table says the max is set to chunk size by default.
This is the code I am talking about

if @fbitTailBufferMaxSizeMBs > 0 && @fbitTailBufferChunkSizeMBs == 0
          @fbitTailBufferChunkSizeMBs = @fbitTailBufferMaxSizeMBs
          puts "config::warn: since tailbufchunksizemegabytes not provided hence using tailbufchunksizemegabytes=#{@fbitTailBufferChunkSizeMBs} which is same as the value of tailbufmaxsizemegabytes"
        elsif  @fbitTailBufferChunkSizeMBs > 0  && @fbitTailBufferMaxSizeMBs == 0
          @fbitTailBufferMaxSizeMBs = @fbitTailBufferChunkSizeMBs
          puts "config::warn: since tailbufmaxsizemegabytes not provided hence using tailbufmaxsizemegabytes=#{@fbitTailBufferMaxSizeMBs} which is same as the value of tailbufchunksizemegabytes"
        end

Reply:
just verified the fbit crashes if we only set tailbufchunksizemegabytes but not max one.

replacing nodename in telegraf config
File Doesnt Exist. Creating file...
←[1mFluent Bit v1.4.2←[0m
* ←[1m←[93mCopyright (C) 2019-2020 The Fluent Bit Authors←[0m
* ←[1m←[93mCopyright (C) 2015-2018 Treasure Data←[0m
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[engine] caught signal (SIGSEGV)
2021-01-08T22:04:12Z I! Starting Telegraf
Telegraf unknown (git: fork 50cd124)
/opt/main.sh: line 592:   439 Aborted                 (core dumped) /opt/td-agent-bit/bin/td-agent-bit -c /etc/opt/microsoft/docker-cimprov/td-agent-bit.conf -e /opt/td-agent-bit/bin/out_oms.so

Reply:
> > 1. It is the same documentation for tail plugin you shared earlier - https://fluentbit.io/documentation/0.13/input/tail.html
> >    Look under the table for the settings.
> > 2. In your code "if @fbitTailBufferMaxSize > 0 && @fbitTailBufferChunkSize == 0
> >    @fbitTailBufferChunkSize = @fbitTailBufferMaxSize"
> >    You are setting the chunk size to max size if it is not set. This is also in the same tail documentation.
> 
> > The doc only talking out the default value. do you know if buf_max_size will be buf_chunk_size if not proided? as I pointed out here, I think, if the value of the buf_chunk_size   is greater than buf_max_size  then the fbit will fail with an error https://github.com/fluent/fluent-bit/blob/master/plugins/in_tail/tail_config.c#L165. Let me do quick validation and see what's behavior.

>  just verified. fbit crashes for example if chunksize ="20"  and similarly crashes if we specify  both chunksize=20 and maxsize =1 but not met the criterion  maxsize > chunksize .